backend/src/eventHandlers/resignHandler.ts

```diff
@@ -0,0 +1,21 @@
+import type { RoomData, BoardState } from "../types";
+
+exports = function (
+  socket: any,
+  io: any,
+  username: string,
+  rooms: Map<string, RoomData>,
+  boards: Map<string, BoardState>
+) {
+  // CLIENT WORKFLOW FOR RESIGNATION
+  // client A should emit "resign" if they attempt to resign
+  // all resignations are successful and both clients should handle "game resigned" event with the username of client A being sent
+  // the server will handle deletion of rooms
+  socket.on("resign", () => {
```

Comment:
What would be the convention to call this function in the Chess UI when say, the client presses the resign button?

Reply:
It would be socket.emit("resign"). There is more in-depth documentation on the socket.io website.